### PR TITLE
Android UI: lots of cleanups, mainly around how Fragments are handled

### PR DESCRIPTION
**Pull request by hackbar**
This has a lot of stuff in it, but it's mostly around better abstraction with the emulation lifecycle, and using more standard Android features (eg animators).

### CODE REVIEW

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/fragments/MenuFragment.java [hunk 1]
```diff
@@ -42,7 +53,11 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
 			button.setOnClickListener(this);
 		}
 
-		mTitleText = (TextView) rootView.findViewById(R.id.text_game_title);
+		TextView titleText = rootView.findViewById(R.id.text_game_title);
+		String title = getArguments().getString(KEY_TITLE);
+		if (title != null){
```
Comment (lioncash) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/fragments/MenuFragment.java]: brace should go on the next line.

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/nVidiaShieldWorkaroundView.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import android.content.Context;
+import android.util.AttributeSet;
+import android.view.View;
+
+/**
+ * Work around a bug with the nVidia Shield.
+ */
+public final class nVidiaShieldWorkaroundView extends View
+{
+
```
Comment (lioncash) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/nVidiaShieldWorkaroundView.java]: unnecessary newline

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/nVidiaShieldWorkaroundView.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+package org.dolphinemu.dolphinemu.ui;
+
+import android.content.Context;
+import android.util.AttributeSet;
+import android.view.View;
+
+/**
+ * Work around a bug with the nVidia Shield.
+ */
+public final class nVidiaShieldWorkaroundView extends View
```
Comment (lioncash) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/nVidiaShieldWorkaroundView.java]: Classes should generally start with an upper-case letter.

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/EmulationActivity.java [hunk 4]
```diff
@@ -463,12 +398,13 @@ private void showScreenshot()
 		@Override
 		public void run()
 		{
-			mFrameContent.removeView(mFrameEmulation);
 			setResult(mPosition);
 			finishAfterTransition();
 		}
 	};
 
+
```
Comment (lioncash) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/activities/EmulationActivity.java]: Unnecessary newlines?

**Comment by hackbar** [comment]
Thanks for the review; I've fixed all these.

**Comment by hackbar** [comment]
Hold off on merging this, I just found a bug with rotation and the controls overlay. I don't know if it's new, but it should be fixed.

**Comment by hackbar** [comment]
I've rebased and added one more commit to fix a minor bug where the input overlay would race between Wii or Gamecube. There's still a bug with the overlays disappearing on rotation, but considering rotation is completely broken in HEAD, this change is still a positive.

I'm going to keep working at that, but I think it might be a framework bug. For the meantime, can you take another look at this?

**Comment by hackbar** [comment]
@degasus can you merge this?

File: Source/Android/app/build.gradle [hunk 5]
```diff
@@ -1,8 +1,8 @@
 apply plugin: 'com.android.application'
 
 android {
-    compileSdkVersion 25
-    buildToolsVersion '25.0.2'
+    compileSdkVersion 26
+    buildToolsVersion '26.0.1'
```
[the remark this answers is not in the document]
Comment (hackbar) [reply Source/Android/app/build.gradle]: Probably update after this PR was created. Updated.

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/main/TvMainActivity.java [hunk 6]
```diff
@@ -33,11 +34,11 @@
 import org.dolphinemu.dolphinemu.utils.StartupHandler;
 import org.dolphinemu.dolphinemu.viewholders.TvGameViewHolder;
 
-public final class TvMainActivity extends Activity implements MainView
+public final class TvMainActivity extends FragmentActivity implements MainView
```
[the remark this answers is not in the document]
Comment (hackbar) [reply Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/ui/main/TvMainActivity.java]: No, that would just make things more complicated, not simpler. Just pass around the base Activity (be is AppCompat* or Fragment*, depending on if you have to interact with leanback [which has no AppCompat]).

**Comment by Etualbadusername** [comment]
How does one use the android version?

**Pull request merged**